src/Pulsar.Client/Internal/ClientCnx.fs

```diff
@@ -212,73 +209,75 @@ type ClientCnx (broker: Broker,
-                            then
-                                Log.Logger.LogError("Error: {0}. Message: {1}", cmd.Error, cmd.Message)
-                                Error
-                            else
-                                LookupTopicResult { BrokerServiceUrl = cmd.brokerServiceUrl; Proxy = cmd.ProxyThroughServiceUrl }
-                        handleRespone %cmd.RequestId result reader consumed
-                    | XCommandProducerSuccess (cmd, consumed) ->
-                        let result = ProducerSuccess { GeneratedProducerName = cmd.ProducerName }
-                        handleRespone %cmd.RequestId result reader consumed
-                    | XCommandSuccess (cmd, consumed) ->
-                        handleRespone %cmd.RequestId Empty reader consumed
-                    | XCommandCloseProducer (cmd, consumed) ->
-                        let producerMb = producers.[%cmd.ProducerId]
-                        producers.Remove(%cmd.ProducerId) |> ignore
-                        producerMb.Post(ProducerMessage.ConnectionClosed)
-                        reader.AdvanceTo(consumed)
-                    | XCommandCloseConsumer (cmd, consumed) ->
-                        let consumerMb = consumers.[%cmd.ConsumerId]
-                        consumers.Remove(%cmd.ConsumerId) |> ignore
-                        consumerMb.Post(ConsumerMessage.ConnectionClosed)
-                        reader.AdvanceTo(consumed)
-                    | XCommandReachedEndOfTopic (cmd, consumed) ->
-                        let consumerMb = consumers.[%cmd.ConsumerId]
-                        consumerMb.Post ReachedEndOfTheTopic
-                        reader.AdvanceTo(consumed)
-                    | XCommandGetTopicsOfNamespaceResponse (cmd, consumed) ->
-                        let result = TopicsOfNamespace { Topics = List.ofSeq cmd.Topics }
-                        handleRespone %cmd.RequestId result reader consumed
-                    | XCommandError (cmd, consumed) ->
-                        Log.Logger.LogError("Error: {0}. Message: {1}", cmd.Error, cmd.Message)
-                        let result = Error
-                        handleRespone %cmd.RequestId result reader consumed
-                    | IncompleteCommand ->
+                    | Result.Error IncompleteCommand ->
                         reader.AdvanceTo(buffer.Start, buffer.End)
-                    | InvalidCommand ex ->
-                        raise ex
+                    | Result.Error (UnknownCommandType unknownType) ->
+                        failwithf "Unknown command type %A" unknownType
+                    | Result.Ok (xcmd, consumed) ->
+                        match xcmd with
+                        | XCommandConnected _ ->
+                            //TODO check server protocol version
+                            initialConnectionTsc.SetResult(this)
+                            reader.AdvanceTo consumed
```

Comment:
Common consumed logic can now be extracted once for all cases

Reply:
right now I'm not that involved to know what the common logic here is. Maybe it is more suitable for another PR

Reply:
I mean `reader.AdvanceTo consumed` line which can be extracted for all success cases to the bottom of match